plugins/sonarqube/src/index.ts

```diff
@@ -21,9 +21,16 @@
  * @packageDocumentation
  */
 
-export * from './components';
+export type {
+  DuplicationRating,
+  SonarQubeContentPageProps,
+  SONARQUBE_PROJECT_KEY_ANNOTATION,
+  isSonarQubeAvailable,
+} from './components';
```

Comment:
as a hint here, [ADR004](https://backstage.io/docs/architecture-decisions/adrs-adr004) suggests it's useful to make "leaf" index files (like `./components/index.ts` in this case) have a list of named exports, ONLY the public ones - so that "non-leaf" index files like this one can safely do star exports. That way it's easy to look inside the components folder to see what's being exported for public use, without having to manually drill upwards in the hierarchy to piece together the puzzle.

So it would be neat if this file could only contain

```ts
export  * from './components';
export  * from './plugin';
```

and then the index files inside were adapted accordingly.

Reply:
Really make sense. Thanks for the hints.